econplayground/main/views.py

```diff
@@ -591,3 +593,168 @@ def get_success_url(self):
             extra_tags='safe')
 
         return reverse('topic_list', kwargs={'cohort_pk': self.cohort.pk})
+
+
+class AssignmentListView(LoginRequiredMixin, ListView):
+    model = Assignment
+    template_name = 'main/assignment_list.html'
+
+    def dispatch(self, request, *args, **kwargs):
+        if not user_is_instructor(request.user):
+            url = '/accounts/login/'
+            return HttpResponseRedirect(url)
+
+        return super(
+            AssignmentListView, self).dispatch(request, *args, **kwargs)
+
+    def get_queryset(self):
+        return Assignment.objects.filter(
+            instructors=self.request.user).order_by('created_at')
+
+    def get_context_data(self, **kwargs):
+        context = super(AssignmentListView, self).get_context_data(**kwargs)
+
+        # This area should never be reached if this user isn't an instructor,
+        # but just in case.
+        if not user_is_instructor(self.request.user):
+            return context
+
+        # Create a clone of the sample course if this user has no
+        # courses.
+        if not Assignment.objects.filter(
+                instructors__in=(self.request.user,)).exists():
+            try:
+                sample_assignment = Assignment.objects.get(is_sample=True)
+            except Assignment.DoesNotExist:
+                sample_assignment = None
+
+            if sample_assignment:
+                cloned_sample = sample_assignment.clone()
+                cloned_sample.instructors.clear()
+                cloned_sample.instructors.add(self.request.user)
+
+        return context
+
+
+class AssignmentDetailView(DetailView):
+    model = Assignment
+
+    def get_graph_queryset(self):
+        # First, set up graphs based on a users role
+        questions = Question.objects.filter()
+
+        # Then apply filtering based on query string params
+        params = self.request.GET
+        if len(params) == 0:
+            return questions.order_by('created_at')
+        return questions.order_by('title')
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        question_list = self.get_graph_queryset()
+
+        context['question_list'] = question_list
+        context['all_count'] = question_list.count()
+
+        return context
+
+
+class AssignmentCreateView(
+        EnsureCsrfCookieMixin, UserPassesTestMixin, CreateView):
+    model = Assignment
+    fields = ['title']
+
+    def test_func(self):
+        return user_is_instructor(self.request.user)
+
+    def get_success_url(self):
+        return '/assignment'
```

Comment:
Rather than hard-coding the URL path, it's best practice in django to name the url path in urls.py, and refer to it using the dynamic [reverse](https://docs.djangoproject.com/en/4.1/ref/urlresolvers/#reverse) method.

Reply:
Makes sense.